pkg/auth/defaultmanager_test.go

```diff
@@ -28,24 +32,70 @@ func (s *TestDefaultManagerSuite) TestKeyManagerDefaultKeyManager() {
 	assert.True(s.T(), s.Config.IsTestingMode(), "testing mode not set correctly to true")
 
 	s.Run("get before init", func() {
-		_, err := auth.DefaultKeyManager()
+		_, err := DefaultKeyManager()
 		require.Error(s.T(), err)
 		require.Equal(s.T(), "no default KeyManager created, call `InitializeDefaultKeyManager()` first", err.Error())
 	})
 
 	s.Run("first creation", func() {
-		_, err := auth.InitializeDefaultKeyManager(logger, s.Config)
+		_, err := InitializeDefaultKeyManager(logger, s.Config)
 		require.NoError(s.T(), err)
 	})
 
 	s.Run("second redundant creation", func() {
-		_, err := auth.InitializeDefaultKeyManager(logger, s.Config)
+		_, err := InitializeDefaultKeyManager(logger, s.Config)
 		require.Error(s.T(), err)
 		require.Equal(s.T(), "default KeyManager can be created only once", err.Error())
 	})
 
 	s.Run("retrieval", func() {
-		_, err := auth.DefaultKeyManager()
+		_, err := DefaultKeyManager()
 		require.NoError(s.T(), err)
 	})
+
+	s.Run("parallel threads", func() {
+		// reset the singleton
+		resetDefaultKeyManager()
+		type kmErrHolder struct {
+			KeyMngr *KeyManager
+			KmErr   error
+		}
+		latch := sync.WaitGroup{}
+		latch.Add(1)
+		holder := make([]*kmErrHolder, 3)
+		for i := 0; i < 3; i++ {
+			go func(i int) {
+				// now, wait for latch to be released so that all workers start at the same time
+				latch.Wait()
+				km, err := InitializeDefaultKeyManager(logger, s.Config)
+				thisHolder := &kmErrHolder{
+					KeyMngr: km,
+					KmErr:   err,
+				}
+				holder[i] = thisHolder
+			}(i)
+		}
+		latch.Done()
+		// wait a few cycles before checking the results
+		time.Sleep(time.Millisecond * 1000)
+		// check if only one entry has a KeyManager and the two others have errs
+		fmt.Println(holder)
+		fails := 0
+		success := 0
+		for i := 0; i < 3; i++ {
+			thisEntry := holder[i]
+			fmt.Println(thisEntry.KmErr)
```

Comment:
please remove this line, or use `t.Logf(...)` so it'll be printed if the test fails

Reply:
fixed 5876bb9.